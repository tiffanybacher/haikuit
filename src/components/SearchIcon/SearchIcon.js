import React from 'react';

function SearchIcon(props) {
  return (
    <button 
      className="SearchIcon" 
      onClick={props.toggleSearch}
      tabindex={props.tabindex}>
      <i className="fas fa-search"></i>
    </button>
  );
}

export default SearchIcon;