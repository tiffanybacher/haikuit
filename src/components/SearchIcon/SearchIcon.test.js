import React from 'react';
import { shallow } from 'enzyme';
import SearchIcon from './SearchIcon';

describe('SearchIcon', () => {
  let toggleSearch;
  let tabindex;
  let wrapper;

  beforeEach(() => {
    toggleSearch = jest.fn();
    tabindex = jest.fn();
    wrapper = shallow(
      <SearchIcon 
        toggleSearch={toggleSearch}
        tabindex={tabindex} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should show search bar when clicked', () => {
    wrapper.find('.SearchIcon').simulate('click');

    expect(toggleSearch).toHaveBeenCalled();
  });
});