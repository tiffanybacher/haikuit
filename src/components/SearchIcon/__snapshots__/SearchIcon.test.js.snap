// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchIcon should match snapshot 1`] = `
<button
  className="SearchIcon"
  onClick={[MockFunction]}
  tabindex={[MockFunction]}
>
  <i
    className="fas fa-search"
  />
</button>
`;
